intelmq/bots/experts/sieve/expert.py

```diff
@@ -256,64 +279,91 @@ def process_ip_range_match(self, key, ip_range, event):
+        rhs = value.values
+        result = lhs == rhs if op == ':equals' else self._list_op_map[op](set(lhs), set(rhs))
+
+        return not result if neg else result
+
+    def process_bool_match(self, key, op, value, event):
+        if not (key in event and isinstance(event[key], bool)):
+            return False
+
+        return self._bool_op_map[op](event[key], value)
+
+    def compute_basic_math(self, action, event) -> str:
         date = DateTime.parse_utc_isoformat(event[action.key], True)
-        if action.operator == '+=':
-            return (date + datetime.timedelta(minutes=parse_relative(action.value))).isoformat()
-        elif action.operator == '-=':
-            return (date - datetime.timedelta(minutes=parse_relative(action.value))).isoformat()
+        delta = datetime.timedelta(minutes=parse_relative(action.value))
 
-    @staticmethod
-    def process_action(action, event):
+        return self._basic_math_op_map[action.operator](date, delta).isoformat()
+
+    def process_action(self, action, event) -> Procedure:
+        name = action.__class__.__name__
         if action == 'drop':
             return Procedure.DROP
         elif action == 'keep':
             return Procedure.KEEP
-        elif action.__class__.__name__ == 'PathAction':
+        elif name == 'PathAction':
             event.path = action.path
-        elif action.__class__.__name__ == 'AddAction':
+        elif name == 'AddAction':
             if action.key not in event:
                 value = action.value
                 if action.operator != '=':
-                    value = SieveExpertBot.compute_basic_math(action, event)
+                    value = self.compute_basic_math(action, event)
                 event.add(action.key, value)
-        elif action.__class__.__name__ == 'AddForceAction':
+        elif name == 'AddForceAction':
             value = action.value
             if action.operator != '=':
-                value = SieveExpertBot.compute_basic_math(action, event)
+                value = self.compute_basic_math(action, event)
             event.add(action.key, value, overwrite=True)
-        elif action.__class__.__name__ == 'UpdateAction':
+        elif name == 'UpdateAction':
             if action.key in event:
                 value = action.value
                 if action.operator != '=':
-                    value = SieveExpertBot.compute_basic_math(action, event)
+                    value = self.compute_basic_math(action, event)
                 event.change(action.key, value)
-        elif action.__class__.__name__ == 'RemoveAction':
+        elif name == 'RemoveAction':
             if action.key in event:
                 del event[action.key]
+        elif name == 'AppendAction':
+            if action.key in event and isinstance(event[action.key], list):
+                list_ = event[action.key]
+                list_.append(action.value)
+                event.add(action.key, list_, overwrite=True)
```

Comment:
Is the `overwrite=True` intentioned, even though this is not the force-flavor of the Append-Action?

Reply:
Now that I look at it, the `event.add` line in the non-force variant is currently redundant as `event[action.key]` and `list_` are the same list at the same memory locations, and updating `_list` implicitly updates the list at `event[action.key]`.
The same line is not redundant in `AppendForceAction` as a new list may be created. Though for this same reason, if `AppendAction` were to be made to also optionally create a new list if the key is not already included in the event, this line would be there to handle the case of a new list being created.
An alternative would be to simply only have the `event.add` in the branch of code where a new list is created.